### PR TITLE
Update and vendor various dependencies prior to a new release

### PR DESCRIPTION
**Pull request by kwilczynski**
#### What type of PR is this?

/kind dependency-change
/assign kwilczynski

#### What this PR does / why we need it:

Update various direct build time dependencies prior to a new release. This change also closes several dependency changes that had been open for a while.

Related:

- https://github.com/cri-o/cri-o/pull/7364
- https://github.com/cri-o/cri-o/pull/7368
- https://github.com/cri-o/cri-o/pull/7385
- https://github.com/cri-o/cri-o/pull/7431
- https://github.com/cri-o/cri-o/pull/7451
- https://github.com/cri-o/cri-o/pull/7453
- https://github.com/cri-o/cri-o/pull/7487
- https://github.com/cri-o/cri-o/pull/7489
- https://github.com/cri-o/cri-o/pull/7496
- https://github.com/cri-o/cri-o/pull/7516
- https://github.com/cri-o/cri-o/pull/7523
- https://github.com/cri-o/cri-o/pull/7537
- https://github.com/cri-o/cri-o/pull/7545
- https://github.com/cri-o/cri-o/pull/7547
- https://github.com/containers/conmon-rs/pull/1713
- https://github.com/containers/podman/pull/19873

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

See review comments.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,29 +7,29 @@ require (
...
 	github.com/containerd/cri-containerd v1.19.0
 	github.com/containerd/fifo v1.1.0
-	github.com/containerd/nri v0.3.1-0.20230504231226-94185418e253
+	github.com/containerd/nri v0.5.0
 	github.com/containerd/ttrpc v1.2.2
 	github.com/containerd/typeurl v1.0.3-0.20220422153119-7f6e6d160d67
 	github.com/containernetworking/cni v1.1.2
 	github.com/containernetworking/plugins v1.3.0
-	github.com/containers/common v0.55.4
+	github.com/containers/common v0.57.0
```
Comment (kwilczynski) [inline go.mod:22]: This change means that the following Pull Request is no longer needed:

- https://github.com/cri-o/cri-o/pull/7496

File: go.mod [hunk 2]
```diff
@@ -7,29 +7,29 @@ require (
 	github.com/Microsoft/go-winio v0.6.1
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/blang/semver/v4 v4.0.0
-	github.com/checkpoint-restore/checkpointctl v0.1.0
-	github.com/checkpoint-restore/go-criu/v6 v6.3.0
-	github.com/container-orchestrated-devices/container-device-interface v0.6.0
+	github.com/checkpoint-restore/checkpointctl v1.1.0
+	github.com/checkpoint-restore/go-criu/v7 v7.0.0
```
Comment (kwilczynski) [inline go.mod:11]: This change follows the following upstream update:

- https://github.com/containers/podman/pull/19873

And, when included, would resolve the following build time error:

```
panic: proto: file "stats/stats.proto" is already registered
	previously from: "github.com/checkpoint-restore/go-criu/v6/stats"
	currently from:  "github.com/checkpoint-restore/go-criu/v7/stats"
```

File: go.mod [hunk 3]
```diff
@@ -7,29 +7,29 @@ require (
...
+	github.com/containerd/nri v0.5.0
 	github.com/containerd/ttrpc v1.2.2
 	github.com/containerd/typeurl v1.0.3-0.20220422153119-7f6e6d160d67
 	github.com/containernetworking/cni v1.1.2
 	github.com/containernetworking/plugins v1.3.0
-	github.com/containers/common v0.55.4
+	github.com/containers/common v0.57.0
 	github.com/containers/conmon v2.0.20+incompatible
-	github.com/containers/conmon-rs v0.6.1
+	github.com/containers/conmon-rs v0.6.2-0.20230920142715-f5a362044a57
```
Comment (kwilczynski) [inline go.mod:24]: This change follows the following upstream change:

- https://github.com/containers/conmon-rs/pull/1713

And, when included, would resolve the following build time error:

```
# github.com/containers/conmon-rs/pkg/client
vendor/github.com/containers/conmon-rs/pkg/client/attach.go:232:19: undefined: util.CopyDetachable
vendor/github.com/containers/conmon-rs/pkg/client/attach.go:429:26: undefined: util.ErrDetach
```

Comment (kwilczynski) [reply go.mod:24]: To add: this vendors a specific commit for the time being in anticipation of a future new release of conmon-rs.

File: go.mod [hunk 4]
```diff
@@ -7,29 +7,28 @@ require (
 	github.com/Microsoft/go-winio v0.6.1
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/blang/semver/v4 v4.0.0
-	github.com/checkpoint-restore/checkpointctl v0.1.0
-	github.com/checkpoint-restore/go-criu/v6 v6.3.0
-	github.com/container-orchestrated-devices/container-device-interface v0.6.0
```
Comment (kwilczynski) [inline go.mod:12]: This change follows the following upstream update:

- https://github.com/cncf-tags/container-device-interface/pull/172

And, when included, would resolve package import issues.

**Comment by kwilczynski** [comment]
> The linter is not happy, though:
> 
> ```
>   Error: SA1019: otelgrpc.UnaryServerInterceptor is deprecated: Use [NewServerHandler] instead.  (staticcheck)
>   Error: SA1019: otelgrpc.StreamServerInterceptor is deprecated: Use [NewServerHandler] instead.  (staticcheck)
> ```

@saschagrunert, we can fix this later in a follow-up Pull Request.

This might be a larger body of work, and I don't want to mix dependency updates with code refactorings, if possible.

Unless you want this fixed here?

**Comment by saschagrunert** [comment]
@kwilczynski I'd prefer to fix it within this PR, you can just go for a second commit on top of the dependency changes.

**Comment by kwilczynski** [comment]
> @kwilczynski I'd prefer to fix it within this PR, you can just go for a second commit on top of the dependency changes.

@saschagrunert, OK! Will do.

**Review by saschagrunert — APPROVED** [review]

**Comment by haircommander** [comment]
@klihub it looks like nri tests are failing: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/cri-o_cri-o/7526/pull-ci-cri-o-cri-o-main-ci-fedora-integration/1731673898904195072/artifacts/fedora-integration/cri-o-gather/artifacts/testout.txt is it due to the update?

**Comment by klihub** [comment]
@haircommander @kwilczynski Can you guys pull in this commit/fix: https://github.com/klihub/cri-o/commit/53f55e39b514d09caf41bd20f0078d773c34ee1c

That should give

```
Bats 1.9.0
++ bats --jobs 8 --tap nri.bats
1..12
ok 1 run NRI PluginRegistration test
ok 2 run NRI PluginSynchronization test
ok 3 run NRI PodEvents test
ok 4 run NRI ContainerEvents test
ok 5 run NRI MountInjection test
ok 6 run NRI EnvironmentInjection test
ok 7 run NRI AnnotationInjection test
ok 8 run NRI DeviceInjection test
ok 9 run NRI CpusetAdjustment test
ok 10 run NRI MemsetAdjustment test
ok 11 run NRI CpusetAdjustmentUpdate test
ok 12 run NRI MemsetAdjustmentUpdate test
```

**Review by klihub — CHANGES_REQUESTED** [review]

File: go.mod [hunk 5]
```diff
@@ -7,29 +7,28 @@ require (
...
+	github.com/checkpoint-restore/checkpointctl v1.1.0
+	github.com/checkpoint-restore/go-criu/v7 v7.0.0
 	github.com/containerd/cgroups v1.1.0
-	github.com/containerd/containerd v1.7.5
+	github.com/containerd/containerd v1.7.10
 	github.com/containerd/cri-containerd v1.19.0
 	github.com/containerd/fifo v1.1.0
-	github.com/containerd/nri v0.3.1-0.20230504231226-94185418e253
+	github.com/containerd/nri v0.5.0
```
[the remark this answers is not in the document]
Comment (klihub) [reply go.mod:16]: And I apologize for not having spotted this in time/earlier....

[the remark this answers is not in the document]
Comment (kwilczynski) [reply go.mod:16]: @klihub, thank you for taking care of this so promptly! Much appreciated :)

**Comment by kwilczynski** [comment]
> @haircommander @kwilczynski Can you guys pull in this commit/fix: [klihub@53f55e3](https://github.com/klihub/cri-o/commit/53f55e39b514d09caf41bd20f0078d773c34ee1c)
> 
> That should give
> 
> ```
> Bats 1.9.0
> ++ bats --jobs 8 --tap nri.bats
> 1..12
> ok 1 run NRI PluginRegistration test
> ok 2 run NRI PluginSynchronization test
> ok 3 run NRI PodEvents test
> ok 4 run NRI ContainerEvents test
> ok 5 run NRI MountInjection test
> ok 6 run NRI EnvironmentInjection test
> ok 7 run NRI AnnotationInjection test
> ok 8 run NRI DeviceInjection test
> ok 9 run NRI CpusetAdjustment test
> ok 10 run NRI MemsetAdjustment test
> ok 11 run NRI CpusetAdjustmentUpdate test
> ok 12 run NRI MemsetAdjustmentUpdate test
> ```

Done. Completed in https://github.com/cri-o/cri-o/pull/7526/commits/c80c3aa5f5d557a6fd790f64dca77bc8ae9800fd.

Thank you a bunch for help, @klihub! :+1:

**Comment by haircommander** [comment]
hm need a [docs validation update](https://github.com/cri-o/cri-o/actions/runs/7089610098/job/19294847996?pr=7526)

**Comment by kwilczynski** [comment]
> hm need a [docs validation update](https://github.com/cri-o/cri-o/actions/runs/7089610098/job/19294847996?pr=7526)

@haircommander, will take care of it.

**Comment by kwilczynski** [comment]
I also took the liberty of including the following update:

- https://github.com/cri-o/cri-o/pull/7547

Especially as our tireless bot has already noticed a recent update to this dependency.

**Review by sohankunkerkar — APPROVED** [review]
/lgtm

**Pull request merged**